django/db/migrations/autodetector.py

```diff
@@ -861,19 +860,13 @@ def generate_altered_fields(self):
             old_field_dec = self.deep_deconstruct(old_field)
             new_field_dec = self.deep_deconstruct(new_field)
             if old_field_dec != new_field_dec:
-                both_m2m = (
-                    isinstance(old_field, models.ManyToManyField) and
-                    isinstance(new_field, models.ManyToManyField)
-                )
-                neither_m2m = (
-                    not isinstance(old_field, models.ManyToManyField) and
-                    not isinstance(new_field, models.ManyToManyField)
-                )
+                both_m2m = old_field.many_to_many and new_field.many_to_many
```

Comment:
I'd love to see that tested against [django-taggit](https://github.com/alex/django-taggit) once that supports Django >= 1.10